### ISSUE TITLE
Refactor: extract special Expression type 

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#5256]
So we can link it to the parameter type page just like we do with color.

**Comment by adrijshikhar** [comment]
@kanitw is this issue still open?

**Comment by kanitw** [comment]
Yes, right now all expression just have type strings, which are not distinguished from arbitrary strings.  

It'd be nice to do treat them slightly different, just like what we do with `Color`.

**Comment by adrijshikhar** [comment]
Thanks for your reply. This is my first issue, so can you guide me a bit on what is required. I have already setup the repo.

**Comment by kanitw** [comment]
I'll search for the type Color and see how it's being used.  Then you'd want to do the same thing with expression for properties listed above.
